Frameworks/CoreText/CTFontManager.mm

```diff
@@ -34,53 +34,102 @@ CFArrayRef CTFontManagerCreateFontDescriptorsFromURL(CFURLRef fileURL) {
     return StubReturn();
 }
 
+// Helper functions to reduce code duplication, as (un)registering fonts have almost identical path
+// (Un)Registers font by creating temporary CFArray and calling into function for multiple fonts
+// TLambda :: (CFArrayRef -> CTFontManagerScope -> CFArrayRef*) -> bool
+template <typename TLambda>
+static bool __CTFontManagerUpdateWithFont(CFURLRef fontURL, CTFontManagerScope scope, CFErrorRef _Nullable* error, TLambda&& func) {
+    woc::unique_cf<CFArrayRef> fontURLs{ CFArrayCreate(nullptr, (const void**)&fontURL, 1, &kCFTypeArrayCallBacks) };
+    if (error) {
```

Comment:
You don't need to double the logic, just call with the errors, log it, if the caller passed in error set it and return